### ISSUE TITLE
[supervision][feature][help] Minimal display routines

### ISSUE THREAD

**Issue by Fabrizio-Caruso** [cc65/cc65#677]
I will implement some minimal graphics routines to display sprites/characters.
I will need them for my universal 8-bit game CROSS CHASE (for about 150 different hardware configurations with any 8-bit CPU from the '70s to the '90s).

I will be more than happy if someone else would use my code as a starting point to write a fully fledged conio and/or tgi driver.
Due to the universal nature of my game, the game's engine is tile-based. Therefore I am implementing some pseudo 20x20 text mode where I could display both some ASCII and some game-specific tiles).
I will write these routines in ANSI C.

Any suggestion, advice and help are welcome.

**Comment by Fabrizio-Caruso** [comment]
How can I get/create an ASCII set of characters where each pixel is described by 2 bits?
I may try CharPad.
Any suggestion on how to configure the linker to have a graphics are where the tiles/characters are stored?

**Comment by polluks** [comment]
Just use another target as template, for example the NES.

**Comment by Fabrizio-Caruso** [comment]
@polluks, thanks.
I would like to just implement some minimal display in C. I am not fluent in any Assembly. 
I just want to display characters in pseudo 20x20 text mode by reading the graphics data from somewhere in memory.

I am trying to the set the world record for the most ported game (with the same game code)... I may have already beaten it :-P
 
I will try to see how far I can go by following the NES template!

**Comment by Fabrizio-Caruso** [comment]
@polluks , @oliverschmidt 
How can I include a character set into the Supervision target?
If I look at the NES cfg, I do not understand how the CHARS segment is filled:
"
MEMORY {
...
    ROM0:   file = %O, start = $8000, size = $7FFA, fill = yes, define = yes;
    # Hardware Vectors at End of 2nd 8K ROM
    ROMV:   file = %O, start = $FFFA, size = $0006, fill = yes;
    # 1 8k CHR Bank
    ROM2:   file = %O, start = $0000, size = $2000, fill = yes;
...
    SRAM:   file = "", start = $0500, size = __STACKSIZE__, define = yes;
...
    RAM:    file = "", start = $6000, size = $2000, define = yes;
}
SEGMENTS {
...
    CHARS:    load = ROM2,            type = rw;
...
}
```
I don't find something similar in the PCE target.

**Comment by oliverschmidt** [comment]
Maybe I don't understand your question...

`ROM2:   file = %O, start = $0000, size = $2000, fill = yes;` defines the memory area.

`CHARS:    load = ROM2,            type = rw;` defines the segment.

```
.segment "CHARS"
.include        "neschar.inc"
```
in https://github.com/cc65/cc65/blob/master/libsrc/nes/crt0.s#L179-L181 defines the segment content.

**Comment by oliverschmidt** [comment]
> I don't find something similar in the PCE target.

As far as I understand the NES target makes sure that the font is placed at the desired location as a side effect of loading the binary by placing it at a defined location inside the binary - therefore the linker config magic.

In contrast the PCE target grammatically copies the font to the desired location after loading the binary. See https://github.com/cc65/cc65/blob/master/libsrc/pce/conio.s#L86 - therefore no linker config magic.

**Comment by Fabrizio-Caruso** [comment]
@oliverschmidt you gave me a good hint. 
I cannot commit myself to improving the supervision target but whatever I do for it, I will share it. Maybe I will manage to contribute something acceptable. Any progress on this target is good.
Since we know now that the joystick driver works and that the display output is straightforward,
the missing brick for the construction of either conio or tgi is a charset stored somewhere and a routine that displays a character on the screen by reading data from the charset.

**Comment by polluks** [comment]
@Fabrizio-Caruso Indeed, the sv target was rather incomplete, it should be something like gamate.

**Comment by Fabrizio-Caruso** [comment]
@polluks Gamate has (almost) working TGI (I may open an issue). Supervision has nothing except a constant to the start address of the memory mapped video ram, which is easy to use (linearly mapped with 4 undisplayed pixels at the right edge of each line).
